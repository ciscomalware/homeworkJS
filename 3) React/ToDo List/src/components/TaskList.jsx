import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { editTask } from '../actions';
import Form from './Form';

const TaskContainer = ({
  task,
}) => {
  const dispatch = useDispatch();

  const handleClickChangeMode = () => {
    dispatch(enterEditMode({ id: task.id }));
  };

  return (
    <>
      <div className="tasks-list-title">{task.title}</div>
      <div className="tasks-list-body">{task.body}</div>
      <button type="button" onClick={handleClickChangeMode}>Редактироват</button>
    </>
  );
};

const TaskList = () => {
  const tasks = useSelector((state) => Object.values(state.tasks));

  const dispatch = useDispatch();

  const handleEditTask = (id) => (formValues) => {
    dispatch(editTask({
      task: {
        ...formValues,
        id,
      },
    }));
  };

  return (
    <div className="tasks">
      <h1 className="tasks-title" />
      <div className="tasks-filter" />
      <div className="tasks-list">
        {tasks.map((task) => (
          <div key={task.id} className="tasks-list-item">
             {task.isEditMode
               ? <Form submitTaskForm={handleEditTask(task.id)} className="edit-task" initialValues={task} />
               : <TaskContainer task={task} />
             }
           </div>
        ))}
      </div>
    </div>
  );
};

export default TaskList;
