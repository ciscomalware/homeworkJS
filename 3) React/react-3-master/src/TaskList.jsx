import React from 'react';

export default class TaskList extends React.Component {
  render() {
    const liElements = this.props.tasks.map((el) => (
      <li className={el.state === 'finished' ? 'list-group-item bg-success' : 'list-group-item bg-primary'} key={el.id}>
        <h5>{el.state === 'finished' ? <s>{el.title}</s> : el.title}</h5>
        <p>{el.state === 'finished' ? <s>{el.body}</s> : el.body}</p>
      </li>));

    return (
        <ul className="list-group-flush">
            {liElements}
        </ul>
    );
  }
}
