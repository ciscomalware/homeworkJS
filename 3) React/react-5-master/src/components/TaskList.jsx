import React from 'react';

export default class TaskList extends React.Component {
  render() {
    return (
      <div className="row">
        {this.props.tasks.map((task) => (task.state === 'finish'
          ? <div className="col-3" key={task.id}>
            <div className="card m-2 text-white bg-primary">
              <div className="card-header">
                <h3 className="card-title">{task.title}</h3>
              </div>
              <div className="card-body">
                <p className="card-text">{task.body}</p>
              </div>
              <div className="card-footer">
                <button type="button" className="btn btn-outline-light" onClick={this.props.finish(task.id)}>Завершить задачу</button>
              </div>
            </div>
          </div>
          : <div className="col-3" key={task.id}>
            <div className="card m-2 text-white bg-danger" key={task.id}>
              <div className="card-header">
                <h3 className="card-title">{task.title}</h3>
              </div>
              <div className="card-body">
                <p className="card-text">{task.body}</p>
              </div>
              <div className="card-footer">
                <button type="button" className="btn btn-outline-light" onClick={this.props.delete(task.id)}>Удалить задачу</button>
              </div>
            </div>
          </div>))}
      </div>
    );
  }
}
