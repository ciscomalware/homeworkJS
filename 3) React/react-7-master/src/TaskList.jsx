import React from 'react';
import { connect } from 'react-redux';
import * as actionCreators from './actions';

class TaskList extends React.Component {
  delete = (id) => () => {
    const { removePost } = this.props;
    removePost({ id });
  }

  render() {
    const { posts } = this.props;
    const postToArray = Object.values(posts);
    return (
      <ul className="list-group-flush">
        {postToArray.map((task) => (
          <li className="list-group-item" key={task.id}>
            <h3>{task.text}</h3>
          <button type="button" className="btn btn-danger" onClick={this.delete(task.id)}>Удалить</button>
        </li>
        ))}
      </ul>
    );
  }
}

const mapStateToProps = (state) => ({ posts: state.posts });

export default connect(mapStateToProps, actionCreators)(TaskList);
