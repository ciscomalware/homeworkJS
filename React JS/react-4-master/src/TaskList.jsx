import React from 'react';
import { uniqueId } from 'lodash';

// TaskList component
export default class TaskList extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      tasks: [
        {
          id: uniqueId(), title: 'Learn React', body: 'I should learn react', state: 'current',
        },
        {
          id: uniqueId(), title: 'Learn JS', body: 'Yay!', state: 'finished',
        },
        {
          id: uniqueId(), title: 'Learn HTML', body: 'DONE!', state: 'finished',
        },
        {
          id: uniqueId(), title: 'Learn Redux', body: 'Should learn after React', state: 'current',
        },
      ],
    };
  }

  activeFn = (id) => () => {
    this.setState({
      tasks: this.state.tasks.map((task) => (task.id === id ? { ...task, state: 'current' } : task))
    })
  }

  finishFn = (id) => () => {
    this.setState({
      tasks: this.state.tasks.map((task) => (task.id === id ? { ...task, state: 'finished' } : task))
    })
  }

  render() {
    const liElements = this.state.tasks.map((el) => (
      <li className={el.state === 'finished' ? 'list-group-item bg-success' : 'list-group-item bg-primary'} key={el.id}>
        <h5>{el.state === 'finished' ? <s>{el.title}</s> : el.title}</h5>
        <p>{el.state === 'finished' ? <s>{el.body}</s> : el.body}</p>
        <button type="button" className="btn btn-info" onClick={this.finishFn(el.id)}>Завершить задачу</button>
        <button type="button" className="btn btn-danger" onClick={this.activeFn(el.id)}>Сделать активной</button>
      </li>));

    return (
        <ul className="list-group-flush">
            {liElements}
        </ul>
    );
  }
}

// import React from 'react';

// export default class Counter extends React.Component {
//   constructor(props) {
//     super(props);
//     this.state = {
//       users: this.props.users,
//     };
//   }

//   removeUser = (id) => () => {
//     this.setState({
//       users: this.state.users.filter((user) => user.id !== id),
//     });
//   }

//   render() {
//     const { users } = this.state;
//     return (
//       <ul className="list-group list-group-flush">
//         {users.map((user) => (
//           <li className="list-group-item" key={user.id}>
//             <button type="button" className="btn btn-danger mr-5"
//            onClick={this.removeUser(user.id)}>Удалить</button>
//             {user.name}
//           </li>
//         ))}
//       </ul>
//     )
//   }
// }
