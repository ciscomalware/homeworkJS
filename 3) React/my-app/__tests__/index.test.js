import React from 'react';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { configure, shallow } from 'enzyme';

import TaskList from '../src/TaskList';


configure({ adapter: new Adapter() });

test('TaskList', () => {
  const component = shallow(
    <TaskList />,
  );

  expect(component.html()).toMatchSnapshot();
  component.find('li.bg-primary').first().find('button').first().simulate('click');
  expect(component.find('li').first().html()).toMatchSnapshot();
  component.find('li.bg-success').first().find('button').first().simulate('click');
  expect(component.find('li').first().html()).toMatchSnapshot();
  component.find('li.bg-success').first().find('button').last().simulate('click');
  expect(component.find('li').first().html()).toMatchSnapshot();
});