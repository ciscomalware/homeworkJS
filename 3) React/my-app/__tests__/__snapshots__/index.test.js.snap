// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskList 1`] = `"<ul class=\\"list-group-flush\\"><li class=\\"list-group-item bg-primary\\"><h5>Learn React</h5><p>I should learn react</p><button type=\\"button\\" class=\\"btn btn-info\\">Завершить задачу</button><button type=\\"button\\" class=\\"btn btn-danger\\">Сделать активной</button></li><li class=\\"list-group-item bg-success\\"><h5><s>Learn JS</s></h5><p><s>Yay!</s></p><button type=\\"button\\" class=\\"btn btn-info\\">Завершить задачу</button><button type=\\"button\\" class=\\"btn btn-danger\\">Сделать активной</button></li><li class=\\"list-group-item bg-success\\"><h5><s>Learn HTML</s></h5><p><s>DONE!</s></p><button type=\\"button\\" class=\\"btn btn-info\\">Завершить задачу</button><button type=\\"button\\" class=\\"btn btn-danger\\">Сделать активной</button></li><li class=\\"list-group-item bg-primary\\"><h5>Learn Redux</h5><p>Should learn after React</p><button type=\\"button\\" class=\\"btn btn-info\\">Завершить задачу</button><button type=\\"button\\" class=\\"btn btn-danger\\">Сделать активной</button></li></ul>"`;

exports[`TaskList 2`] = `"<li class=\\"list-group-item bg-success\\"><h5><s>Learn React</s></h5><p><s>I should learn react</s></p><button type=\\"button\\" class=\\"btn btn-info\\">Завершить задачу</button><button type=\\"button\\" class=\\"btn btn-danger\\">Сделать активной</button></li>"`;

exports[`TaskList 3`] = `"<li class=\\"list-group-item bg-success\\"><h5><s>Learn React</s></h5><p><s>I should learn react</s></p><button type=\\"button\\" class=\\"btn btn-info\\">Завершить задачу</button><button type=\\"button\\" class=\\"btn btn-danger\\">Сделать активной</button></li>"`;

exports[`TaskList 4`] = `"<li class=\\"list-group-item bg-primary\\"><h5>Learn React</h5><p>I should learn react</p><button type=\\"button\\" class=\\"btn btn-info\\">Завершить задачу</button><button type=\\"button\\" class=\\"btn btn-danger\\">Сделать активной</button></li>"`;
